### ISSUE TITLE
Doesn't initialize properly on sites created with WP CLI

### ISSUE THREAD

**Issue by sbrow** [10up/distributor#639]
**Describe the bug**

The multisite connection doesn't load on sites created with the WP CLI

**Steps to Reproduce**

1. Create WP Multisite Install
2. Install and Network Activate Distributor Plugin.
3. Run `$ wp site create --slug=example --title=Example`
4. Navigate to http://example.example.com/wp-admin/admin.php?page=pull and see the error: "No connections to Pull from, Create One?".

Note: Going to the 'Edit Site' page and clicking 'Save Changes' solves this problem.

**Expected behavior**

Should work the same as when sites are created using the web forms.

**Screenshots**

Error page
![image](https://user-images.githubusercontent.com/6729162/94158619-993d6d80-fe50-11ea-8024-78eaf7f8568b.png)

**Environment information**
 - Device: Virtual Machine: Homestead v9.5.1 on Virtualbox 6.1.12
 - OS: Ubuntu
 - Browser and version: Firefox
 - WordPress version: 5.5.1
 - WP CLI version: 2.4.0

**Comment by helen** [comment]
Just wanted to update here that this is reproducible for both push and pull interfaces on the newly created site, and we're working on identifying the cause.

**Comment by dkotter** [comment]
Testing this out today and I was not able to reproduce this issue. For reference, here's my testing details:
- Distributor 1.6.2
- WordPress 5.6.1
- WP-CLI 2.8.1

I also tested in both an environment that had caching enabled and one that didn't. In all tests, I never got that No Connection message, I was always able to see the proper list.

On the Distributor side, we do have caching around that list of available connections. We reset that cache anytime the `clean_site_cache` hook is fired.

Looking at the WP-CLI site create code, it runs `wpmu_create_blog` (https://github.com/wp-cli/entity-command/blob/master/src/Site_Command.php#L463), which in turn will run `wp_insert_site`. That function will eventually run `clean_blog_cache`, which fires the `clean_site_cache` hook. So in theory, anytime a site is created using that WP-CLI command, our cache should be cleared which means a fresh version of that connections list should be shown when going to that Pull screen.

The one area I see that could be causing problems is in that `clean_blog_cache` function, there's a check to see if the global `$_wp_suspend_cache_invalidation` variable is set and if so, it returns early, before the `clean_site_cache` hook is run. So in that scenario, the cache wouldn't be cleared and that could cause issues with that connection list.

Again, I didn't see this happening when I tested but there may be some scenario I didn't account for that could run into this use case.

If wanted, we could add our cache clearing onto the `wp_insert_site` hook, which should always reliably fire when a new site is created. I think we'd still want to hook onto `clean_site_cache`, for other cache clearing scenarios, which would mean we would end up clearing the cache twice in most scenarios, when creating a new site. That's an acceptable scenario I think if it means we more reliably have that connection list updated properly on new site creation.

But all that said, since I was not able to reproduce the problem, also wonder if maybe this was fixed either in a WP-CLI update, a WordPress update or a Distributor update (unlikely but possible).
